lib/map/security_token/service.rb

```diff
@@ -57,17 +57,27 @@ def parse_and_raise_error(e, icn, application)
 
       def parse_response(response, application, icn)
         response_body = response.body
+        validate_map_token(response_body['access_token'])
 
         {
           access_token: response_body['access_token'],
           expiration: Time.zone.now + response_body['expires_in']
         }
+      rescue JWT::VerificationError => e
+        message = "#{config.logging_prefix} token failed, JWT verification error"
+        Rails.logger.error(message, application:, icn:)
+        raise e
       rescue => e
         message = "#{config.logging_prefix} token failed, response unknown"
         Rails.logger.error(message, application:, icn:)
         raise e, "#{message}, application: #{application}, icn: #{icn}"
       end
 
+      def validate_map_token(encoded_token)
+        public_cert = config.provider_certificate
```

Comment:
We're going to want to cache this call, here is an example how we do it for id.me and login.gov public certs: `lib/sign_in/public_jwks.rb`.